tests/delete/tests.py

```diff
@@ -76,18 +76,26 @@ def test_protect(self):
             "referenced through protected foreign keys: 'A.protect'."
         )
         with self.assertRaisesMessage(ProtectedError, msg):
-            a.protect.delete()
+            try:
+                a.protect.delete()
+            except ProtectedError as exc:
+                self.assertEqual(exc.protected_objects, [a])
+                raise
```

Comment:
You don't need to catch and reraise the exception:
```python
        with self.assertRaisesMessage(ProtectedError, msg) as cm:
            a.protect.delete()
        self.assertEqual(cm.exception.protected_objects, [a])
```

Reply:
Thanks @felixxm